Android/wear/src/main/java/dev/saxionroosters/MainActivity.java

```diff
@@ -1,64 +1,259 @@
+            @Override
+            public void onLayoutInflated(WatchViewStub stub) {
+                listView = (WearableListView) stub.findViewById(R.id.list_view);
+                progressBar = (ProgressBar) stub.findViewById(R.id.progressBar);
+                loadAdapter();
+                setProgressBarVisible(true);
+            }
+        });
+
+
+    }
+
+    private void loadAdapter() {
+        listView.setAdapter(mAdapter);
+        listView.setClickListener(this);
+    }
+
+
+    private void setProgressBarVisible(Boolean visibility) {
+        if (!visibility) {
+            // Hide progressbar
+            progressBar.setVisibility(View.GONE);
+        } else {
+            // Make progressbar visible and change color to white
+            progressBar.setVisibility(View.VISIBLE);
+            progressBar.setIndeterminate(true);
+            progressBar.getIndeterminateDrawable().setColorFilter(0xFFFFFFFF, android.graphics.PorterDuff.Mode.SRC_ATOP);
+        }
+    }
+
+    private String formatDate(String inputDate) {
+        SimpleDateFormat sdf = new SimpleDateFormat("EEEE dd MMMM yyyy");
+        Date date = null;
+        try {
+            date = sdf.parse(inputDate);
+        }catch(Exception ex){
+            ex.printStackTrace();
+        }
+        SimpleDateFormat formatter = new SimpleDateFormat("dd MMM");
+
+        return formatter.format(date);
+    }
+
+
+    @Override
+    public void onClick(WearableListView.ViewHolder viewHolder) {
+        switch (viewHolder.getPosition()) {
+            case 0:
+                //Do something
+                break;
+        }
+    }
+
+    @Override
+    public void onTopEmptyRegionClick() {
+        // Prevent NullPointerException
     }
 
     @Override
-    public void onEnterAmbient(Bundle ambientDetails) {
-        super.onEnterAmbient(ambientDetails);
-        updateDisplay();
+    public void onConnected(Bundle bundle) {
+        Tools.log("Wear client | Connected to Google Api Service");
+        Wearable.DataApi.addListener(googleClient, this);
+
+        // Send a request to the host (phone/tablet) to get new college-data
+        PutDataMapRequest putDataMapReq = PutDataMapRequest.create("/get").setUrgent();
+        putDataMapReq.getDataMap().putString("refresh", "true");
+        putDataMapReq.getDataMap().putLong("time", new Date().getTime());
+
+        PutDataRequest putDataReq = putDataMapReq.asPutDataRequest().setUrgent();
+        Wearable.DataApi.putDataItem(googleClient, putDataReq);
+        Tools.log("Wear client | Refresh request sent to host device");
     }
 
     @Override
-    public void onUpdateAmbient() {
-        super.onUpdateAmbient();
-        updateDisplay();
+    public void onConnectionSuspended(int i) {
+        // do nothing
     }
 
     @Override
-    public void onExitAmbient() {
-        updateDisplay();
-        super.onExitAmbient();
+    public void onConnectionFailed(ConnectionResult connectionResult) {
+        // do nothing
```

Comment:
Helaas niet, dit is nodig om de communicatie tussen watch en host device op te zetten. Is voor zover ik weet geen andere manier voor.